control-plane-operator/controllers/hostedcontrolplane/kas/config.go

```diff
@@ -122,15 +122,17 @@ func generateConfig(p KubeAPIServerConfigParams) *kcpv1.KubeAPIServerConfig {
 			},
 			CORSAllowedOrigins: corsAllowedOrigins(p.AdditionalCORSAllowedOrigins),
 		},
-		AuthConfig: kcpv1.MasterAuthConfig{
-			OAuthMetadataFile: cpath(kasVolumeOauthMetadata().Name, OauthMetadataConfigKey),
-		},
 		ConsolePublicURL:             p.ConsolePublicURL,
 		ImagePolicyConfig:            imagePolicyConfig(p.InternalRegistryHostName, p.ExternalRegistryHostNames),
 		ProjectConfig:                projectConfig(p.DefaultNodeSelector),
 		ServiceAccountPublicKeyFiles: []string{cpath(kasVolumeServiceAccountKey().Name, pki.ServiceSignerPublicKey)},
 		ServicesSubnet:               strings.Join(p.ServiceNetwork, ","),
 	}
+
+	if p.Authentication != nil && p.Authentication.Type == configv1.AuthenticationTypeOIDC {
```

Comment:
I might have misunderstood, but I thought we only did this if the type was the internal OIDC or empty.

Reply:
wow yes... the opposite of this please :man_facepalming: